src/sc2laddercore/LadderManager.cpp

```diff
@@ -264,26 +264,53 @@ std::string LadderManager::GetBotCommandLine(const BotConfig &AgentConfig, int G
 	{
 	case Python:
 	{
-		OutCmdLine = "python " + AgentConfig.FileName;
+		OutCmdLine = Config->GetValue("PythonBinary") + " " + AgentConfig.FileName;
+		OutCmdLine += " --GamePort " + std::to_string(GamePort) + " --StartPort " + std::to_string(StartPort) + " --LadderServer 127.0.0.1 --OpponentId " + OpponentId;
+		break;
+	}
+	case Wine:
+	{
+		OutCmdLine = "wine " + AgentConfig.FileName;
+		OutCmdLine += " --GamePort " + std::to_string(GamePort) + " --StartPort " + std::to_string(StartPort) + " --LadderServer 127.0.0.1 --OpponentId " + OpponentId;
+		break;
+	}
+	case Mono:
+	{
+		OutCmdLine = "mono " + AgentConfig.FileName;
+		OutCmdLine += " --GamePort " + std::to_string(GamePort) + " --StartPort " + std::to_string(StartPort) + " --LadderServer 127.0.0.1 --OpponentId " + OpponentId;
+		break;
+	}
+	case CommandCenterLinux:
+	{
+		OutCmdLine = "wine " + Config->GetValue("CommandCenterPath") + " --ConfigFile " + AgentConfig.FileName;
+		OutCmdLine += " --GamePort " + std::to_string(GamePort) + " --StartPort " + std::to_string(StartPort) + " --LadderServer 127.0.0.1 --OpponentId " + OpponentId;
+		break;
+	}
+	case DotNetCore:
+	{
+		OutCmdLine = "dotnet " + AgentConfig.FileName;
+		OutCmdLine += " --GamePort " + std::to_string(GamePort) + " --StartPort " + std::to_string(StartPort) + " --LadderServer 127.0.0.1 --OpponentId " + OpponentId;
 		break;
 	}
 	case BinaryCpp:
 	{
 		OutCmdLine = AgentConfig.RootPath + AgentConfig.FileName;
+		OutCmdLine += " --GamePort " + std::to_string(GamePort) + " --StartPort " + std::to_string(StartPort) + " --LadderServer 127.0.0.1 --OpponentId " + OpponentId;
 		break;
 	}
 	case CommandCenter:
 	{
 		OutCmdLine = Config->GetValue("CommandCenterPath") + " --ConfigFile " + AgentConfig.FileName;
+		OutCmdLine += " --GamePort " + std::to_string(GamePort) + " --StartPort " + std::to_string(StartPort) + " --LadderServer 127.0.0.1 --OpponentId " + OpponentId;
 		break;
 
 	}
 	case DefaultBot:
 	{
-
+		OutCmdLine += " --GamePort " + std::to_string(GamePort) + " --StartPort " + std::to_string(StartPort) + " --LadderServer 127.0.0.1 --OpponentId " + OpponentId;
 	}
 	}
-	OutCmdLine += " --GamePort " + std::to_string(GamePort) + " --StartPort " + std::to_string(StartPort) + " --LadderServer 127.0.0.1 --OpponentId " + OpponentId;
+
```

Comment:
Btw, which executable will be used in this case? Seems that it diesn’t work.

Reply:
This might be an accidental move of the code. I looks like it should be back where it was. I will review this and fix it up.

Reply:
I made the BinaryCpp and DefaultBot types act the same for now (previously DefaultBot did nothing).

Reply:
Actually, I was wrong to do this. DefaultBot had no command line because it is the BlizzardAI and therefore didn't need it - I have reverted the change.